File: src/main/java/xyz/ivyxjc/PropertiesProvider.java
package xyz.ivyxjc;

import java.io.IOException;
import java.io.InputStream;
import java.util.Properties;

public class PropertiesProvider {

    private static Properties props;

    static {
        props = loadProperties("pojo-generator.properties");
    }

    private PropertiesProvider() {
    }

    public static String getBasePackageName() {
        return props.getProperty("pojo_generator.basePackageName");
    }

    public static String getCatalog() {
        return props.getProperty("pojo_generator.catalog");
    }

    public static String getSchema() {
        return props.getProperty("pojo_generator.schema");
    }

    public static String getTable() {
        return props.getProperty("pojo_generator.tableNamePattern");
    }

    public static String getTables() {
        return props.getProperty("pojo_generator.tables");
    }

    public static Properties loadProperties(String resourceName) {
        Properties properties = new Properties();
        InputStream is =
                PropertiesProvider.class.getClassLoader().getResourceAsStream(resourceName);
        if (is == null) {
            return properties;
        }
        try {
            properties.load(is);
        } catch (IOException e) {
            e.printStackTrace();
        } finally {
            if (is != null) {
                try {
                    is.close();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
            return properties;
        }
    }
}

File: src/main/kotlin/xyz/ivyxjc/pojogenerator/PropertiesProvider.kt
package xyz.ivyxjc.pojogenerator

import java.util.*

object PropertiesProvider {

    private var props: Properties? = null

    val packageName: String
        get() = props!!.getProperty("pojo_generator.packageName")

    val schema: String
        get() = props!!.getProperty("pojo_generator.schema")

    val table: String
        get() = props!!.getProperty("pojo_generator.tableNamePattern")

    val catalog: String?
        get() = props?.getProperty("pojo_generator.catalog")

    val tables: String?
        get() = props?.getProperty("pojo_generator.tables")

    val baseClass: String?
        get() = props?.getProperty("pojo_generator.baseClassName")

    val baseColumnNames: String?
        get() = props?.getProperty("pojo_generator.baseColumnNames")

    init {
        props = loadProperties("pojo-generator.properties")
    }

    fun loadProperties(resourceName: String): Properties {
        val properties = Properties()
        ClassLoader.getSystemResourceAsStream(resourceName)?.use { content -> properties.load(content);return properties }
                ?: return properties
    }
}
